lib/lexer/src/lexer.rs

```diff
@@ -406,24 +406,29 @@ impl Lexer {
     fn consume_digits(&mut self, digit_characters: String, token: &mut String) {
         loop {
             // Try consume '_' digit.
-            let character = self.peek_char(None);
-            if character.is_some() && character.unwrap() == '_' {
+            let mut character = self.peek_char(None);
+            while character.is_some() && character.unwrap() == '_' {
                 // Consume '_'.
                 self.eat_char();
+                character = self.peek_char(None);
 
-                // If '_' is consumed, a digit must follow.
-                let character = self.peek_char(None);
-                if character.is_some() && digit_characters.find(character.unwrap()).is_some() {
-                    token.push(self.eat_char());
-                } else { // Otherwise spit out '_' and break.
```

Comment:
I see you've removed this part which spits out consumed underscores if not followed by [digits](https://github.com/astrolang/astro/pull/18/files#diff-90fc61b8161deb5595449d1aaac262cfL414). 

The point is that if you have already consumed underscores and the next set of characters aren't supported digits, then we must regurgitate the underscores we consumed. That is it must be as if we didn't consume the underscores in the first place so that subsequent calls can consume the underscores as appropriate.

The problem with your current approach is that for a code`0b10__name` the underscores will be consumed as part of *binary_literal*, when in fact it should be `0b1` for the *binary_literal* and `__name` for the identifier